source/_posts/2016-01-25-upgrading-to-ruby-on-rails-5-0.markdown

```diff
@@ -0,0 +1,226 @@
+
+
+## Problem 5 - render partial of .html files escapes the content
+
+One case that our tests don’t cover: The header and footer show escaped on the screen (that’s to say, we see these templates’ source code directly in the browser!).
+
+Our header and footer are partials with the extension .html, not .html.erb. After debugging, I notice that the [Raw handler](https://github.com/rails/rails/blob/v5.0.0.beta1/actionview/lib/action_view/template/handlers/raw.rb) is being used to manage those partials and the Raw handler, as seen in the code, returns a String. The render partial is therefore returning a String instead of an OutputBuffer to the process of renderization of the application layout. However, the application layout assembles the HTML to be rendered as an OutputBuffer upon concatenating the String received from the render partial. Since a String is not HTML safe, all that partial’s content is left escaped, thus generating the problem in question.
+
+This began happening after [this commit](https://github.com/rails/rails/commit/4be859f0fdf7b3059a28d03c279f03f5938efc80). Previously, the ERB handler processed an .html file and returned an OutputBuffer. As the default handler is now the Raw handler, the Raw handler processes it and returns a String. This doesn’t mean that the change is completely incorrect, it simply means that changing the default handler presents this detail that must be fixed. 
+
+Not only were the .html files being processed by the ERB handler, but [also the .js files](https://github.com/rails/rails/issues/2394), among others. Naturally, it makes little sense to use the ERB handler in flat JavaScript files and it’s for these cases that [the Raw handler was incorporated](https://github.com/rails/rails/commit/8bea607265a2c9bb9bb2188b0a79089ca373b814).
+
+The issue is that the render of an .html file should always return an OutputBuffer, since by definition the content is HTML safe. Regardless of the context, be it render template, partial, or whatever, it’s because of that that [we introduced a new HTML handler](https://github.com/rails/rails/commit/8a998b0fa7523c2c8eb6d0cf56e40408bf6e9b2e).
+
+With this commit, after a `bundle update rails`, the footer appears in the correct way and the header “improves”, but now it shows the ERB code on the page instead of interpreting it. The problem is that the header contains ERB code but the name of the file is  `_header.html`. In Rails 4.2.5, this worked well because by default the ERB Handler processed it. In Rails 5.0, prior to our change, the Raw handler processed it and now the HTML handler processes it because the file ends in .html and the latter is the correct behavior. In our case, the incorrect thing is that the name of the file is `_header.html` and it contains ERB code. The name should have always been `_header.html.erb`. We rename the file and the header begins to be rendered in the correct manner.
```

Comment:
`In our case, the incorrect thing is that the name of the file is `_header.html` and it contains ERB code. The name should have always been '_header.html.erb'.` => `In our case, the incorrect thing is that the name of the file is `_header.html` but should've been '_header.html.erb' in order to have the ERB handler processing it.`